eventuate-tram-reactive-consumer-jdbc/src/main/java/io/eventuate/tram/reactive/consumer/jdbc/ReactiveSqlTableBasedDuplicateMessageDetector.java

```diff
@@ -0,0 +1,53 @@
+package io.eventuate.tram.reactive.consumer.jdbc;
+
+import io.eventuate.common.jdbc.EventuateDuplicateKeyException;
+import io.eventuate.common.jdbc.EventuateSchema;
+import io.eventuate.common.spring.jdbc.reactive.EventuateSpringReactiveJdbcStatementExecutor;
+import io.eventuate.tram.consumer.common.reactive.ReactiveDuplicateMessageDetector;
+import io.eventuate.tram.messaging.common.SubscriberIdAndMessage;
+import org.reactivestreams.Publisher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.transaction.reactive.TransactionalOperator;
+import reactor.core.publisher.Mono;
+
+public class ReactiveSqlTableBasedDuplicateMessageDetector implements ReactiveDuplicateMessageDetector {
+  private Logger logger = LoggerFactory.getLogger(getClass());
+
+  private EventuateSchema eventuateSchema;
+  private String currentTimeInMillisecondsSql;
+  private TransactionalOperator transactionalOperator;
```

Comment:
@ArtemSidorkin please consider this point.

Reply:
@kwonglau thanks for pointing this out. Since TransactionalOperator is Spring specific this class should be in a Spring specific module. At some point there should be an interface. However, I'd prefer to wait until we start work on supporting another implementation and its clear what the interface should be.